src/engine/engine.rs

```diff
@@ -465,6 +465,9 @@ pub trait StateDiff {
     /// the new_state argument as this method should always return the new values
     /// for any properties that are inconsistent.
     fn diff(&self, new_state: &Self) -> Self::Diff;
+
+    /// Return all contained values unchanged.
+    fn unchanged(&self) -> Self::Diff;
```

Comment:
Is this for optimization purposes? To avoid running unnecessary calculations?

Reply:
Because diffing cached() with cached() looks a bit silly: https://github.com/stratis-storage/stratisd/pull/3288/commits/7cd97525c0daeeec16919c105908705f4ad3f805#diff-39f0bed1717b84fc076fa34b429732430038740d516da7e9b39cf963aec11318R918 .

Reply:
That makes sense!